lib/sda/node.go

```diff
@@ -313,8 +333,47 @@ func (n *Node) DispatchChannel(msgSlice []*SDAData) error {
 	return nil
 }
 
-// DispatchMsg will dispatch this SDAData to the right instance
-func (n *Node) DispatchMsg(sdaMsg *SDAData) error {
+// DispatchMsg takes a message and puts it into a queue for later processing.
+// This allows a protocol to have a backlog of messages.
+func (n *Node) DispatchMsg(msg *SDAData) {
+	dbg.Lvl3(n.Myself(), "Received message")
+	n.msgDispatchQueueMutex.Lock()
+	n.msgDispatchQueue = append(n.msgDispatchQueue, msg)
+	dbg.Lvl3(n.Myself(), "DispatchQueue-length is", len(n.msgDispatchQueue))
+	if len(n.msgDispatchQueue) == 1 && len(n.msgDispatchQueueWait) == 0 {
+		n.msgDispatchQueueWait <- true
+	}
+	n.msgDispatchQueueMutex.Unlock()
+}
+
+func (n *Node) dispatchMsgReader() {
+	for {
```

Comment:
Needs to check if the `msgDispatchClose` is still available or not. Else messages could be dispatched even though the node is closed. Probably a boolean is better instead of a channel, then set the boolean to true and send a msgDispatchQueueWait.